app/policies/run_policy.rb

```diff
@@ -40,4 +40,9 @@ def user_admin?
   def user_coach?
     user && user.coach?
   end
+
+  def problem_visible?
+    record.problem.runs_visible
```

Comment:
Please, use `record.problem.runs_visible?`. It makes it clear that this property returns a boolean value.